<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Project-1: STING-seq</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rotation2</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="project_1.html">Project-1</a>
    </li>
    <li>
      <a href="project_2.html">Project-2</a>
    </li>
  </ul>
</li>
<li>
  <a href="misc.html">Misc.</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/chenh19/rotation2">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Project-1: STING-seq</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>rotation2/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220607code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220607)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220607code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220607)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomchenh19rotation2tree7229c17d9b6463f1bbc4594fa102438cd51c196etargetblank7229c17a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/chenh19/rotation2/tree/7229c17d9b6463f1bbc4594fa102438cd51c196e" target="_blank">7229c17</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomchenh19rotation2tree7229c17d9b6463f1bbc4594fa102438cd51c196etargetblank7229c17a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/chenh19/rotation2/tree/7229c17d9b6463f1bbc4594fa102438cd51c196e" target="_blank">7229c17</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   .RData
    Modified:   .Rhistory

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/project_1.Rmd</code>) and HTML
(<code>docs/project_1.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/7229c17d9b6463f1bbc4594fa102438cd51c196e/analysis/project_1.Rmd" target="_blank">7229c17</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-15
</td>
<td>
wflow_publish("./analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/f0e98f9508e9596c23509444fcbdd3313884d113/docs/project_1.html" target="_blank">f0e98f9</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/e0aa02286b1de1ce5a1f3380cf8a91796d394b8b/analysis/project_1.Rmd" target="_blank">e0aa022</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("./analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/eafa16b18d2d69ef72cf9a590cd9735de366a594/docs/project_1.html" target="_blank">eafa16b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/69b29f12b017538632c517bec183e3c263deee9e/analysis/project_1.Rmd" target="_blank">69b29f1</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("./analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/dfd60ceffbcb855df38e366958785d18327b58ad/docs/project_1.html" target="_blank">dfd60ce</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/49f192280f51683d6b8e7a6d4781346c5c0f6c8d/analysis/project_1.Rmd" target="_blank">49f1922</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("./analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/fd7271e37ee28b6962a733eadcd6f40c5a8cf715/docs/project_1.html" target="_blank">fd7271e</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/1b4d12eaa59ae72221f1aafb9a63cf88c956f12c/docs/project_1.html" target="_blank">1b4d12e</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/a6c402d82138d982d9cddfbbb7c10dddeb89b077/docs/project_1.html" target="_blank">a6c402d</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/cedad99d7128f1858561f48c0e68e0cb61cca09a/docs/project_1.html" target="_blank">cedad99</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/6b3f0212399d5ebb9f8c54aa820cf343c890a9fc/analysis/project_1.Rmd" target="_blank">6b3f021</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/e3b9788cc821f0596a5fc064dee05c13139d131d/docs/project_1.html" target="_blank">e3b9788</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/aed5eeda365f1a8462b6ba0876b451d41a78a027/docs/project_1.html" target="_blank">aed5eed</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/c552123f1547c0e6fe97992e57a5ea0f43f7a457/analysis/project_1.Rmd" target="_blank">c552123</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/45bb6ed9e591f779218416995c51ea3ae9cff951/docs/project_1.html" target="_blank">45bb6ed</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/81fdf42910bf7eb03cdc259c413aaae00f0ec20c/analysis/project_1.Rmd" target="_blank">81fdf42</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/c23765ad794f1472105be1c08b4e5f754e908aa9/docs/project_1.html" target="_blank">c23765a</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/b5fb71c05cf6a6394b5ebbe35ce3f02dcb9765ec/docs/project_1.html" target="_blank">b5fb71c</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/10f6641f857979ff8807651d9e1e905d65104442/analysis/project_1.Rmd" target="_blank">10f6641</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/152325b8f0497089b0606cdf45d41882e7481b80/docs/project_1.html" target="_blank">152325b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/1739dd9ddcabc8874f0b7b8b3ba4db174dd6bd26/analysis/project_1.Rmd" target="_blank">1739dd9</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/543ef4ce5bbf5446217f1f22e477e02637c3def4/docs/project_1.html" target="_blank">543ef4c</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/9b6cb27b945d00fdd5a03457a9a545b33ae417b3/docs/project_1.html" target="_blank">9b6cb27</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/d8908c00757a891b07b43c23e7d4d8715246ff3e/analysis/project_1.Rmd" target="_blank">d8908c0</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/8674c8a9a6e6bc110d708af6fb4ac739b6a7fe4e/docs/project_1.html" target="_blank">8674c8a</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/2972ce63fd772aac3b3f23af310497253d59163d/analysis/project_1.Rmd" target="_blank">2972ce6</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/ada40689d5d530940188e556d01243576f568eda/docs/project_1.html" target="_blank">ada4068</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/7c5402e05e4778c2a1f956f2946b830cf4344122/analysis/project_1.Rmd" target="_blank">7c5402e</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/0d08121af6de5c963439458d6bcfca00052070a6/docs/project_1.html" target="_blank">0d08121</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/62265260b4c243b4379c0d005c92b0f52163a3f9/analysis/project_1.Rmd" target="_blank">6226526</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/dd2046d2fea37ae6af7cdd5c2ab13a64510f4a6d/docs/project_1.html" target="_blank">dd2046d</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/71f5d04b24ce0e82c516ea37c3020a590113c615/analysis/project_1.Rmd" target="_blank">71f5d04</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/2c4cab10af4f07ab13753947786063ea50e7ff30/docs/project_1.html" target="_blank">2c4cab1</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/6e73c04daaf7c44fd6e84e29078b0c9a113f0ba0/analysis/project_1.Rmd" target="_blank">6e73c04</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/d46eaabd80df885047911b79b6b42f6172b9a877/docs/project_1.html" target="_blank">d46eaab</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/be56d9d15e9099284a19bc35e3fa66aa00981af9/analysis/project_1.Rmd" target="_blank">be56d9d</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/02a0c264e314a0cd7a228bc0acc9fc2e2ddb143b/docs/project_1.html" target="_blank">02a0c26</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/e2c15c06fbe78a126991912499dfb21f1d646dc7/analysis/project_1.Rmd" target="_blank">e2c15c0</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/abad46eba81ffb8e27212f8f00fba98bf7271d7b/docs/project_1.html" target="_blank">abad46e</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/68948e3fc40d709954ebe2173d3aea27252ee6a4/analysis/project_1.Rmd" target="_blank">68948e3</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/741027be3ff4b6d47fa575bee0c3d8034eb944b9/docs/project_1.html" target="_blank">741027b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/065d7e9310f57078c1367ca76861c4e3bf616268/analysis/project_1.Rmd" target="_blank">065d7e9</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/bb1581208f588d9182465387911ceed0d6c3a7cc/docs/project_1.html" target="_blank">bb15812</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/b6e09934030981fc8946fc73003c26812d1857a6/analysis/project_1.Rmd" target="_blank">b6e0993</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/93a27ae37969ef2a24fc6c42b5ee251bd65163ef/docs/project_1.html" target="_blank">93a27ae</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/b4a7331e2f2a78bfd5851ed24a61f04934eb1ed6/analysis/project_1.Rmd" target="_blank">b4a7331</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/27121a950fc0f543213211458ddccc9324378218/docs/project_1.html" target="_blank">27121a9</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/fce1ffd29665a824be2be7737e8062aaa451a085/analysis/project_1.Rmd" target="_blank">fce1ffd</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-14
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/44517c1a4a22bc35cb33d30e7ddd431def3acc58/docs/project_1.html" target="_blank">44517c1</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/cc6a40ad4b7df51cee564a5ddb9004c4e3630404/analysis/project_1.Rmd" target="_blank">cc6a40a</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/da08f11d5563d9b2e80248da690acad003cbb4fc/docs/project_1.html" target="_blank">da08f11</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/05ccc35b165fd022f8bd54df760d32aee0ff74f9/analysis/project_1.Rmd" target="_blank">05ccc35</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/572f6ba0fd63908c0e2953800a55e69fbcbf5f9c/docs/project_1.html" target="_blank">572f6ba</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/b8870d354e3e15eb1394215e5a02e107780ebe7a/docs/project_1.html" target="_blank">b8870d3</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/719925e73275bdacd980661782b7af236652885a/docs/project_1.html" target="_blank">719925e</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/e7541fa9954c460a40e4234daeebf354b40dce27/docs/project_1.html" target="_blank">e7541fa</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/9d9615dfcae34e6a448e7f01da10b2c93c4e3b86/docs/project_1.html" target="_blank">9d9615d</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/04feaa73f8ed707991ee387e647b667d21a43f33/analysis/project_1.Rmd" target="_blank">04feaa7</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/bbd8978428e806d0c8d30ffdc4ecc75ea6b40560/docs/project_1.html" target="_blank">bbd8978</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/0ec2bfaa748ca34233500d7de0d0e102880eac9b/analysis/project_1.Rmd" target="_blank">0ec2bfa</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/e5a5b52ac0d8b0b9a278432942ac46685fb97cc5/docs/project_1.html" target="_blank">e5a5b52</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/c43ae1f05baa8d3a79998d7e1d8f2acdc8d9c2d0/analysis/project_1.Rmd" target="_blank">c43ae1f</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/4d8bd729213dcedb9cbf4b5626bc0472f5d58b85/docs/project_1.html" target="_blank">4d8bd72</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/3373521dd223485065b5d3aff9d6525ae2a69baa/docs/project_1.html" target="_blank">3373521</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/af21ea8272be1de714d37f5d5c25b7661d3beb86/docs/project_1.html" target="_blank">af21ea8</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/6e56d75ddafb0a0df6e06845d5c29d70eaf7d761/analysis/project_1.Rmd" target="_blank">6e56d75</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/f653f7bffb8c4fe3d5c31d99b1df21770fb518e4/docs/project_1.html" target="_blank">f653f7b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/2723e7fcd5791efbe646bc452eb2d3b7238ec431/analysis/project_1.Rmd" target="_blank">2723e7f</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/d69c892f26960486d57d77644ea648cfe1140d22/docs/project_1.html" target="_blank">d69c892</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/34d877d03d74eaf8e77248cc22912fb56acc1837/docs/project_1.html" target="_blank">34d877d</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/e72400b8c8afd61ebd6a9407367965b63c5ec2cd/docs/project_1.html" target="_blank">e72400b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/c411223a3cd60d8aa14d3c419a6fac5b85ba98f3/docs/project_1.html" target="_blank">c411223</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/1daccd2d37f0e60b15c87ee1a2b4303c48cfad02/docs/project_1.html" target="_blank">1daccd2</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/63f46d225dc4c360684a9fc89490cbe9d38c80cc/analysis/project_1.Rmd" target="_blank">63f46d2</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/26adb457366115270afeeff90e27daba83af034f/docs/project_1.html" target="_blank">26adb45</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/a6022a8556f02e80567c8de12ca748423986c298/docs/project_1.html" target="_blank">a6022a8</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/1215832bc3d7de62020871cbca5a3b1e51247d4b/analysis/project_1.Rmd" target="_blank">1215832</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/9abc4b853fd0f9a4307d2f13c2575458798663ac/docs/project_1.html" target="_blank">9abc4b8</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/7efcfe03baeb846adde012d3bb3a5227716516b7/analysis/project_1.Rmd" target="_blank">7efcfe0</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/f18d38523588a5b0a4c7eaca84b7ca0675414de3/docs/project_1.html" target="_blank">f18d385</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/a7c1ce0dc7f8f6b1eafb37ed507713e877bbd25f/analysis/project_1.Rmd" target="_blank">a7c1ce0</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/e991f56ded6c30a020d631ffa5a152b3a2e6b35f/docs/project_1.html" target="_blank">e991f56</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/3c9b1d94b149395c0d9dd43f6fda1eb161c1f888/docs/project_1.html" target="_blank">3c9b1d9</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/ae1553ad4873a6b39700e7fbcba4167270061806/analysis/project_1.Rmd" target="_blank">ae1553a</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/34e0d024bd6bf88e4085053af353c236409cff1a/docs/project_1.html" target="_blank">34e0d02</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/e69aa839043cbc1d51c553e6950bcf5df5ec71b9/analysis/project_1.Rmd" target="_blank">e69aa83</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/9be31af2adecb402f1caa7d71cb08ba43d1a4923/docs/project_1.html" target="_blank">9be31af</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/ead84c2592d14f84e7baf5f2577693870707a7c2/analysis/project_1.Rmd" target="_blank">ead84c2</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
wflow_publish("analysis/*.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/0f41de809b415e36cf7e9188d0e4c0ba46165c7f/docs/project_1.html" target="_blank">0f41de8</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/31ad0357e3043fca3f2c2297ebc2b6de53bf118a/docs/project_1.html" target="_blank">31ad035</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/bdf3b441c4fa84cd0da1c22c5b4a8bf7c9e683ef/docs/project_1.html" target="_blank">bdf3b44</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/8d0890cbda8875f90861b9efa51047490be43c80/docs/project_1.html" target="_blank">8d0890c</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/chenh19/rotation2/blob/26f455bb10abfe5f9add7484f4b33a324dfe4a1f/analysis/project_1.Rmd" target="_blank">26f455b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/chenh19/rotation2/26f455bb10abfe5f9add7484f4b33a324dfe4a1f/docs/project_1.html" target="_blank">26f455b</a>
</td>
<td>
chenh19
</td>
<td>
2022-06-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="understand-sting-seq" class="section level2">
<h2>1. Understand STING-seq</h2>
<div id="a.-read-the-morris-paper" class="section level3">
<h3>a. Read the Morris paper</h3>
<p><a
href="https://chenh19.github.io/rotation2/refs/Morris_2020.pdf">Morris
<em>et al.</em> 2021: STING-Seq</a></p>
<div id="some-key-points" class="section level4">
<h4>Some key points:</h4>
<p><strong>Some key ideas:</strong></p>
<ul>
<li>STING-seq: Systematic Targeting and Inbition of Noncoding GWAS loci
with scRNA-seq</li>
<li>prioritizes candidate <em>cis</em>-regulatory elements (cCREs,
1kb&lt;distance to TSS&lt;1Mb) using fine-mapped GWAS</li>
<li>selected 88 variants (in 56 loci) with enhancer activity</li>
<li>dual CRISPR inhibition: dCas9 as the GPS, MeCP2 and KRAB as the
repressors</li>
<li>confirming dual CRISPRi efficacy: gRNAs target TSS of
<em>MRPS23</em>, <em>CTSB</em>, <em>FSCN1</em></li>
<li>CRIPSRi on the 88 variants: two gRNAs for each variant, both within
200bp of the variant</li>
<li>ECCITE-seq: captures gRNAs and epitopes</li>
</ul>
<p><strong>Some data processing steps and results:</strong></p>
<ul>
<li>QC: remove cells with low total reads or excessive mitochondrial
reads, gRNA assignment UMI&gt;5 (9,343 cells after QC)</li>
<li><strong>Kallisto:</strong> counting <a
href="https://pachterlab.github.io/kallisto/">read more on the official
website</a></li>
<li><strong>Seurat:</strong> QC and <code>reference mapping?</code> <a
href="https://satijalab.org/seurat/">read more on the official
website</a></li>
<li><strong>SCEPTRE:</strong> gRNA_to_gene-expression pairwise test</li>
<li>non-targeting gRNA-gene pairs: not significant (negative ctrl)</li>
<li>TSS-targeting gRNA-gene pairs: expression significantly decreased
(positive ctrl)</li>
<li>37 of the 88 variants were significant</li>
<li><em>Trans</em>-regulatory elements:
<code>I'll come back later</code></li>
</ul>
<p><strong>Note:</strong></p>
<ul>
<li><a href="https://cite-seq.com/eccite-seq/">Expanded
CRISPR-compatible CITE-seq (ECCITE-Seq)</a></li>
<li><a href="https://cite-seq.com/">Cellular Indexing of Transcriptomes
and Epitopes by Sequencing (CITE-seq)</a></li>
</ul>
</div>
</div>
</div>
<div id="prelim-qc-for-sting-seq-data" class="section level2">
<h2>2. Prelim QC for STING-seq data</h2>
<div id="a.-read-about-rna-seq-analysis" class="section level3">
<h3>a. Read about RNA-seq analysis</h3>
<p><a
href="https://chenh19.github.io/rotation2/refs/Yalamanchili_2017.pdf">Yalamanchili
<em>et al.</em> 2017: RNA-seq analysis pipeline</a></p>
<div id="some-key-points-1" class="section level4">
<h4>Some key points:</h4>
<p><strong>Protocol-1 (differential expression of genes):</strong></p>
<ul>
<li>demuxed raw reads (FastQC)</li>
<li>trimming reads (awk)</li>
<li>aligning reads (TopHat2)</li>
<li>counting reads (HTSeq; may filter out genes with low counts before
next step)</li>
<li>detect DE using counted reads (DEseq2)</li>
<li>more QC (PCA/correlation heatmap)</li>
</ul>
<p><strong>Protocol-2 (differential usage of isoforms):</strong></p>
<ul>
<li>Protocol-1</li>
<li>counting isoforms (Kallisto, <strong>also check <a
href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">cell
ranger</a></strong>)</li>
<li>detect DU using counted isoforms (Sleuth)</li>
<li>more QC (aslo PCA/correlation heatmap)</li>
</ul>
<p><strong>Protocol-3 (crypic splicing):</strong></p>
<ul>
<li>Protocol-1</li>
<li>detect differential junstions (CrypSplice)</li>
</ul>
</div>
</div>
<div id="b.-download-data" class="section level3">
<h3>b. Download data</h3>
<pre class="bash"><code>rsync -a hchen131@midway2.rcc.uchicago.edu:/project2/xuanyao/data/Morris_2021 ~/Desktop</code></pre>
</div>
<div id="c.-perform-fastqc-on-all-fastq-files" class="section level3">
<h3>c. Perform FastQC on all fastq files</h3>
<pre class="bash"><code># install fastqc
sudo apt-get update &amp;&amp; sudo apt-get install fastqc -y

# run fastqc in parallel
cd ~/rotation2/
[ ! -d ../Morris_2021/fastqc_results/ ] &amp;&amp; mkdir ../Morris_2021/fastqc_results/
fastqc --threads 16 ../Morris_2021/STINGseq_Morris_2021_raw/*.fastq.gz --outdir=../Morris_2021/fastqc_results/</code></pre>
<p><strong>SRR14141135:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141135_cDNA_Homo_sapiens_1_fastqc.html">SRR14141135_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141135_cDNA_Homo_sapiens_2_fastqc.html">SRR14141135_2</a></li>
</ul>
<p><strong>SRR14141136:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141136_cDNA_Homo_sapiens_1_fastqc.html">SRR14141136_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141136_cDNA_Homo_sapiens_2_fastqc.html">SRR14141136_2</a></li>
</ul>
<p><strong>SRR14141137:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141137_cDNA_Homo_sapiens_1_fastqc.html">SRR14141137_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141137_cDNA_Homo_sapiens_2_fastqc.html">SRR14141137_2</a></li>
</ul>
<p><strong>SRR14141138:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141138_cDNA_Homo_sapiens_1_fastqc.html">SRR14141138_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141138_cDNA_Homo_sapiens_2_fastqc.html">SRR14141138_2</a></li>
</ul>
<p><strong>SRR14141139:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141139_cDNA_Homo_sapiens_1_fastqc.html">SRR14141139_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141139_cDNA_Homo_sapiens_2_fastqc.html">SRR14141139_2</a></li>
</ul>
<p><strong>SRR14141140:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141140_cDNA_Homo_sapiens_1_fastqc.html">SRR14141140_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141140_cDNA_Homo_sapiens_2_fastqc.html">SRR14141140_2</a></li>
</ul>
<p><strong>SRR14141141:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141141_cDNA_Homo_sapiens_1_fastqc.html">SRR14141141_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141141_cDNA_Homo_sapiens_2_fastqc.html">SRR14141141_2</a></li>
</ul>
<p><strong>SRR14141142:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141142_cDNA_Homo_sapiens_1_fastqc.html">SRR14141142_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141142_cDNA_Homo_sapiens_2_fastqc.html">SRR14141142_2</a></li>
</ul>
<p><strong>SRR14141143:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141143_HTO_Homo_sapiens_1_fastqc.html">SRR14141143_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141143_HTO_Homo_sapiens_2_fastqc.html">SRR14141143_2</a></li>
</ul>
<p><strong>SRR14141144:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141144_HTO_Homo_sapiens_1_fastqc.html">SRR14141144_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141144_HTO_Homo_sapiens_2_fastqc.html">SRR14141144_2</a></li>
</ul>
<p><strong>SRR14141145:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141145_GDO_Homo_sapiens_1_fastqc.html">SRR14141145_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141145_GDO_Homo_sapiens_2_fastqc.html">SRR14141145_2</a></li>
</ul>
<p><strong>SRR14141146:</strong></p>
<ul>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141146_GDO_Homo_sapiens_1_fastqc.html">SRR14141146_1</a><br />
</li>
<li><a
href="https://chenh19.github.io/rotation2/fastqc_reports/SRR14141146_GDO_Homo_sapiens_2_fastqc.html">SRR14141146_2</a></li>
</ul>
<p><strong>A brief summary:</strong></p>
<ul>
<li><strong>length:</strong> <code>26bp</code> or <code>57bp</code>
(<strong>trimmed?</strong>)</li>
<li><strong>depth:</strong> <code>30-35x</code></li>
<li><strong>overall quality:</strong> good (within
<code>~40 bp</code>)</li>
</ul>
</div>
<div id="d.-kallisto-bustools-pipeline" class="section level3">
<h3>d. Kallisto | bustools pipeline</h3>
<p><strong>Install package: pip3 install kb:</strong></p>
<pre class="bash"><code># install via python3-pip (python3 is installed by default)
sudo apt-get update &amp;&amp; sudo apt install python3-pip -y
sudo pip3 install kb-python

# run
kb</code></pre>
<p><strong>Side note: conda install kallisto (not necessary for
kb):</strong></p>
<pre class="bash"><code># install anaconda
[ ! -d ./shscript/ ] &amp;&amp; mkdir ./shscript/
wget -O ./shscript/Anaconda-latest-Linux-x86_64.sh https://repo.anaconda.com/archive/Anaconda3-2022.05-Linux-x86_64.sh
bash ./shscript/Anaconda-latest-Linux-x86_64.sh &amp;&amp; sleep 3
conda update anaconda -y &amp;&amp; conda update --all -y
conda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge
conda install kallisto -y
conda config --set auto_activate_base false # disable auto activate base in terminal
#conda activate # activate base when needed
#rm -rf ~/anaconda3/ # uninstall anaconda
rm -rf ./shscript/</code></pre>
</div>
</div>
<div id="analyze-qced-sting-seq-data" class="section level2">
<h2>3. Analyze QC’ed STING-seq data</h2>
<p><strong>Note:</strong></p>
<ul>
<li><strong>cDNA</strong>, <strong>HTO</strong> (Hashing
antibody-oligos), <strong>GDO</strong> (gRNAs)</li>
<li><strong>ECCITE-seq:</strong></li>
</ul>
<center>
<img src="https://citeseq.files.wordpress.com/2018/11/eccite.png"
style="width:70.0%" />
</center>
<div id="a.-install-r-packages" class="section level3">
<h3>a. Install R packages</h3>
<pre class="bash"><code>sudo apt-get update &amp;&amp; sudo apt-get install libgeos-dev -y
sudo -i R
installe.packages(&quot;Seurat&quot;)
q()</code></pre>
</div>
<div id="b.-import-qced-data" class="section level3">
<h3>b. Import QC’ed data</h3>
<p><strong>Note: about <a
href="https://slowkow.com/notes/sparse-matrix/">sparse
matrix</a></strong></p>
<pre class="r"><code># load expression data
Expression &lt;- Seurat::ReadMtx(mtx = &quot;../Morris_2021/GSM5225857_cDNA.mtx&quot;, 
                                features = &quot;../Morris_2021/GSM5225857_cDNA.features.txt&quot;, 
                                cells = &quot;../Morris_2021/GSM5225857_cDNA.barcodes.txt&quot;, 
                                cell.column=1, 
                                feature.column=1, 
                                mtx.transpose=T)</code></pre>
<pre class="r"><code># load gRNA data
gRNA &lt;- Seurat::ReadMtx(mtx = &quot;../Morris_2021/GSM5225858_GDO.mtx&quot;, 
                                 features = &quot;../Morris_2021/GSM5225858_GDO.features.txt&quot;, 
                                 cells = &quot;../Morris_2021/GSM5225858_GDO.barcodes.txt&quot;, 
                                 cell.column=1, 
                                 feature.column=1, 
                                 mtx.transpose=T)</code></pre>
<pre class="r"><code>overall &lt;- function(mtx){
  # mtx &lt;- Expression
  
  cat(paste0(&#39;The [&#39;, deparse(substitute(mtx)), &#39;] matrix has: \n&#39;, 
        &#39;- &#39;, format(mtx@Dim[1], big.mark=&quot;,&quot;, scientific=FALSE), &#39; rows/genes/targets and \n&#39;, 
        &#39;- &#39;, format(mtx@Dim[2], big.mark=&quot;,&quot;, scientific=FALSE), &#39; columns/barcodes/cells \n&#39;,
        &#39;- &#39;, format(length(mtx), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values in total \n&#39;, 
        &#39;- &#39;, format(sum(mtx != 0), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are non-zero \n&#39;,
        &#39;- &#39;, format(sum(mtx == 1), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are 1 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 1), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 1 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 10), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 10 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 100), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 100 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 1000), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 1,000 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 10000), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 10,000 \n&#39;,
        &#39;- &#39;, format(sum(mtx &gt; 100000), big.mark=&quot;,&quot;, scientific=FALSE), &#39; values that are bigger than 100,000 \n&#39;,
        &#39; \n&#39;))
}
overall(Expression)
overall(gRNA)</code></pre>
<pre><code>&gt; overall(Expression)
The [Expression] matrix has: 
- 35,606 rows/genes/targets and 
- 686,612 columns/barcodes/cells 
- 24,447,506,872 values in total 
- 82,507,471 values that are non-zero 
- 50,421,358 values that are 1 
- 32,086,113 values that are bigger than 1 
- 3,370,699 values that are bigger than 10 
- 259,734 values that are bigger than 100 
- 2,515 values that are bigger than 1,000 
- 0 values that are bigger than 10,000 
- 0 values that are bigger than 100,000 
 
&gt; overall(gRNA)
The [gRNA] matrix has: 
- 210 rows/genes/targets and 
- 137,347 columns/barcodes/cells 
- 28,842,870 values in total 
- 2,506,474 values that are non-zero 
- 1,510,919 values that are 1 
- 995,555 values that are bigger than 1 
- 121,554 values that are bigger than 10 
- 41,071 values that are bigger than 100 
- 2,232 values that are bigger than 1,000 
- 20 values that are bigger than 10,000 
- 0 values that are bigger than 100,000 
</code></pre>
<p><strong>try:</strong></p>
<pre class="r"><code>class(f_exp_matrix)
attributes(f_exp_matrix)
dim(f_exp_matrix)
f_exp_matrix[1,1]
summary(f_exp_matrix[,1])


col_num &lt;- 199
test &lt;- f_exp_matrix[,col_num]
sum(test !=0)
plot(test)

row_num &lt;- 199

row_num &lt;- 199
test &lt;- f_exp_matrix[row_num,]
sum(test != 0)
plot(test)

sum(f_exp_matrix &gt;= 1)
sum(f_exp_matrix &gt; 1)
sum(f_exp_matrix == 1)</code></pre>
<ul>
<li>gene by cell</li>
<li>check how many cells with non zero expression genes</li>
<li>check how many gene with non zero expression cells</li>
<li>hist()</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     bslib_0.3.1      compiler_4.2.0   pillar_1.7.0    
 [5] later_1.3.0      git2r_0.30.1     jquerylib_0.1.4  tools_4.2.0     
 [9] getPass_0.2-2    digest_0.6.29    jsonlite_1.8.0   evaluate_0.15   
[13] tibble_3.1.7     lifecycle_1.0.1  pkgconfig_2.0.3  rlang_1.0.2     
[17] cli_3.3.0        rstudioapi_0.13  yaml_2.3.5       xfun_0.31       
[21] fastmap_1.1.0    httr_1.4.3       stringr_1.4.0    knitr_1.39      
[25] sass_0.4.1       fs_1.5.2         vctrs_0.4.1      rprojroot_2.0.3 
[29] glue_1.6.2       R6_2.5.1         processx_3.6.0   fansi_1.0.3     
[33] rmarkdown_2.14   callr_3.7.0      magrittr_2.0.3   whisker_0.4     
[37] ps_1.7.0         promises_1.2.0.1 htmltools_0.5.2  ellipsis_0.3.2  
[41] httpuv_1.6.5     utf8_1.2.2       stringi_1.7.6    crayon_1.5.1    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
