---
title: "Project-1: STING-seq"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## 1. Understand STING-seq

### a. Read the Morris paper

[Morris *et al.* 2021: STING-Seq](https://chenh19.github.io/rotation2/refs/Morris_2020.pdf)

#### Some key points:

**Some key ideas:**  

* STING-seq: Systematic Targeting and Inbition of Noncoding GWAS loci with scRNA-seq
* prioritizes candidate *cis*-regulatory elements (cCREs, 1kb<distance to TSS<1Mb) using fine-mapped GWAS
* selected 88 variants (in 56 loci) with enhancer activity
* dual CRISPR inhibition: dCas9 as the GPS, MeCP2 and KRAB as the repressors
* confirming dual CRISPRi efficacy: gRNAs target TSS of *MRPS23*, *CTSB*, *FSCN1*
* CRIPSRi on the 88 variants: two gRNAs for each variant, both within 200bp of the variant
* ECCITE-seq: captures gRNAs and epitopes

**Some data processing steps and results:**  

* QC: remove cells with low total reads or excessive mitochondrial reads, gRNA assignment UMI>5 (9,343 cells after QC)
* **Kallisto:** counting [read more on the official website](https://pachterlab.github.io/kallisto/)
* **Seurat:** QC and ```reference mapping?``` [read more on the official website](https://satijalab.org/seurat/)
* **SCEPTRE:** gRNA_to_gene-expression pairwise test
* non-targeting gRNA-gene pairs: not significant (negative ctrl)
* TSS-targeting gRNA-gene pairs: expression significantly decreased (positive ctrl)
* 37 of the 88 variants were significant
* *Trans*-regulatory elements: ```I'll come back later```

**Note:**  

* [Expanded CRISPR-compatible CITE-seq (ECCITE-Seq)](https://cite-seq.com/eccite-seq/)
* [Cellular Indexing of Transcriptomes and Epitopes by Sequencing (CITE-seq)](https://cite-seq.com/)




## 2. Prelim QC for STING-seq data

### a. Read about RNA-seq analysis

[Yalamanchili *et al.* 2017: RNA-seq analysis pipeline](https://chenh19.github.io/rotation2/refs/Yalamanchili_2017.pdf)  

#### Some key points:

**Protocol-1 (differential expression of genes):**  

* demuxed raw reads (FastQC)
* trimming reads (awk)
* aligning reads (TopHat2)
* counting reads (HTSeq; may filter out genes with low counts before next step)
* detect DE using counted reads (DEseq2)
* more QC (PCA/correlation heatmap)

**Protocol-2 (differential usage of isoforms):**  

* Protocol-1
* counting isoforms (Kallisto, **also check [cell ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger)**)
* detect DU using counted isoforms (Sleuth)
* more QC (aslo PCA/correlation heatmap)

**Protocol-3 (crypic splicing):**  

* Protocol-1
* detect differential junstions (CrypSplice)

### b. Download data
```{bash,eval=FALSE}
rsync -a hchen131@midway2.rcc.uchicago.edu:/project2/xuanyao/data/Morris_2021 ~/Desktop
```

### c. Perform FastQC on all fastq files
```{bash,eval=FALSE}
# install fastqc
sudo apt-get update && sudo apt-get install fastqc -y

# run fastqc in parallel
cd ~/rotation2/
[ ! -d ../Morris_2021/fastqc_results/ ] && mkdir ../Morris_2021/fastqc_results/
fastqc --threads 16 ../Morris_2021/STINGseq_Morris_2021_raw/*.fastq.gz --outdir=../Morris_2021/fastqc_results/
```

**SRR14141135:**  

* [SRR14141135_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141135_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141135_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141135_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141136:**  

* [SRR14141136_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141136_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141136_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141136_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141137:**  

* [SRR14141137_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141137_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141137_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141137_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141138:**  

* [SRR14141138_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141138_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141138_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141138_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141139:**  

* [SRR14141139_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141139_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141139_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141139_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141140:**  

* [SRR14141140_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141140_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141140_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141140_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141141:**  

* [SRR14141141_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141141_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141141_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141141_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141142:**  

* [SRR14141142_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141142_cDNA_Homo_sapiens_1_fastqc.html)  
* [SRR14141142_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141142_cDNA_Homo_sapiens_2_fastqc.html)  
  
**SRR14141143:**  

* [SRR14141143_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141143_HTO_Homo_sapiens_1_fastqc.html)  
* [SRR14141143_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141143_HTO_Homo_sapiens_2_fastqc.html)  
  
**SRR14141144:**  

* [SRR14141144_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141144_HTO_Homo_sapiens_1_fastqc.html)  
* [SRR14141144_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141144_HTO_Homo_sapiens_2_fastqc.html)  
  
**SRR14141145:**  

* [SRR14141145_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141145_GDO_Homo_sapiens_1_fastqc.html)  
* [SRR14141145_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141145_GDO_Homo_sapiens_2_fastqc.html)  
  
**SRR14141146:**  

* [SRR14141146_1](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141146_GDO_Homo_sapiens_1_fastqc.html)  
* [SRR14141146_2](https://chenh19.github.io/rotation2/fastqc_reports/SRR14141146_GDO_Homo_sapiens_2_fastqc.html)  
  
  
**A brief summary:**  

* **length:** ```26bp``` or ```57bp``` (**trimmed?**)
* **depth:** ```30-35x```
* **overall quality:** good (within ```~40 bp```)

### d. Kallisto | bustools pipeline

**Install package: pip3 install kb:**
```{bash,eval=FALSE}
# install via python3-pip (python3 is installed by default)
sudo apt-get update && sudo apt install python3-pip -y
sudo pip3 install kb-python

# run
kb
```

**Side note: conda install kallisto (not necessary for kb):**
```{bash,eval=FALSE}
# install anaconda
[ ! -d ./shscript/ ] && mkdir ./shscript/
wget -O ./shscript/Anaconda-latest-Linux-x86_64.sh https://repo.anaconda.com/archive/Anaconda3-2022.05-Linux-x86_64.sh
bash ./shscript/Anaconda-latest-Linux-x86_64.sh && sleep 3
conda update anaconda -y && conda update --all -y
conda config --add channels defaults
conda config --add channels bioconda
conda config --add channels conda-forge
conda install kallisto -y
conda config --set auto_activate_base false # disable auto activate base in terminal
#conda activate # activate base when needed
#rm -rf ~/anaconda3/ # uninstall anaconda
rm -rf ./shscript/
```


## 3. Analyze QC'ed STING-seq data

**Note:**  

* **cDNA**, **HTO** (Hashing antibody-oligos), **GDO** (gRNAs)
* **ECCITE-seq:**  

<center>
![](https://citeseq.files.wordpress.com/2018/11/eccite.png){width=70%}
</center>

### a. Install R packages

```{bash,eval=FALSE}
sudo apt-get update && sudo apt-get install libgeos-dev -y
sudo -i R
installe.packages("Seurat")
q()
```

### b. Import QC'ed data

**Note: about [sparse matrix](https://slowkow.com/notes/sparse-matrix/)**

```{r,eval=FALSE}
# load expression data
Expression <- Seurat::ReadMtx(mtx = "../Morris_2021/GSM5225857_cDNA.mtx", 
                                features = "../Morris_2021/GSM5225857_cDNA.features.txt", 
                                cells = "../Morris_2021/GSM5225857_cDNA.barcodes.txt", 
                                cell.column=1, 
                                feature.column=1, 
                                mtx.transpose=T)

# load gRNA data
gRNA <- Seurat::ReadMtx(mtx = "../Morris_2021/GSM5225858_GDO.mtx", 
                                 features = "../Morris_2021/GSM5225858_GDO.features.txt", 
                                 cells = "../Morris_2021/GSM5225858_GDO.barcodes.txt", 
                                 cell.column=1, 
                                 feature.column=1, 
                                 mtx.transpose=T)

```


```{r,eval=FALSE}
overall <- function(mtx){
  cat(paste0('The [', deparse(substitute(mtx)), '] matrix has: \n', 
        '- ', format(mtx@Dim[1], big.mark=",", scientific=FALSE), 
        ' rows/genes/targets \n', 
        '- ', format(mtx@Dim[2], big.mark=",", scientific=FALSE), 
        ' columns/barcodes/cells \n', 
        '- ', format(length(mtx), big.mark=",", scientific=FALSE), 
        ' values in total \n', 
        '- ', format(sum(mtx != 0), big.mark=",", scientific=FALSE), 
        ' values that are non-zero \n', 
        '- ', format(sum(mtx == 1), big.mark=",", scientific=FALSE), 
        ' values that are 1 \n', 
        '- ', format(sum(mtx > 1), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 1 \n', 
        '- ', format(sum(mtx > 10), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 10 \n', 
        '- ', format(sum(mtx > 100), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 100 \n', 
        '- ', format(sum(mtx > 1000), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 1,000 \n', 
        '- ', format(sum(mtx > 10000), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 10,000 \n', 
        '- ', format(sum(mtx > 100000), big.mark=",", scientific=FALSE), 
        ' values that are bigger than 100,000 \n', 
        ' \n'))
}

overall(Expression)
overall(gRNA)

```

```
> overall(Expression)
The [Expression] matrix has: 
- 35,606 rows/genes/targets 
- 686,612 columns/barcodes/cells 
- 24,447,506,872 values in total 
- 82,507,471 values that are non-zero 
- 50,421,358 values that are 1 
- 32,086,113 values that are bigger than 1 
- 3,370,699 values that are bigger than 10 
- 259,734 values that are bigger than 100 
- 2,515 values that are bigger than 1,000 
- 0 values that are bigger than 10,000 
- 0 values that are bigger than 100,000 
 
> overall(gRNA)
The [gRNA] matrix has: 
- 210 rows/genes/targets
- 137,347 columns/barcodes/cells 
- 28,842,870 values in total 
- 2,506,474 values that are non-zero 
- 1,510,919 values that are 1 
- 995,555 values that are bigger than 1 
- 121,554 values that are bigger than 10 
- 41,071 values that are bigger than 100 
- 2,232 values that are bigger than 1,000 
- 20 values that are bigger than 10,000 
- 0 values that are bigger than 100,000 

```



```{r,eval=FALSE}
mtx <- gRNA
mtx <- mtx[,1:10000]

target_names <- rownames(mtx)
target_means <- c()
for (target_name in target_names){
  target_means <- c(target_means, mean(mtx[target_name,]))
}
target_means <- data.frame(target_names,target_means)
colnames(target_means) <- c("target_name", "target_mean")

barcode_seqs <- colnames(mtx)
barcode_means <- c()
for (barcode_seq in barcode_seqs){
  barcode_means <- c(barcode_means, mean(mtx[,barcode_seq]))
}
barcode_means <- data.frame(barcode_seqs,barcode_means)
colnames(barcode_means) <- c("barcode_seq", "barcode_mean")

max_target <- target_means[which.max(target_means$target_mean), "target_name"]
min_target <- target_means[which.min(target_means$target_mean), "target_name"]

max_barcode <- barcode_means[which.max(barcode_means$barcode_mean), "barcode_seq"]
min_barcode <- barcode_means[which.min(barcode_means$barcode_mean), "barcode_seq"]


plot(mtx[max_target,])
plot(mtx[,max_barcode])

plot(mtx[max_target,], main="The highest gRNA in all cells", sub="",
  xlab=paste0(max_target, " (gRNA)"), ylab="Expression level")
plot(mtx[min_target,], main="The lowest gRNA in all cells", sub="",
  xlab=paste0(min_target, " (gRNA)"), ylab="Expression level")

plot(mtx[,max_barcode], main="The cell with highest overall gRNAs", sub="",
  xlab=paste0(max_barcode, " (cell)"), ylab="Expression level")
plot(mtx[,min_barcode], main="The cell with lowest overall gRNAs", sub="",
  xlab=paste0(min_barcode, " (cell)"), ylab="Expression level")
```

![]()





